src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/GoogleDataprocDAO.scala

```diff
@@ -0,0 +1,88 @@
+
+
+import scala.collection.JavaConverters._
+import scala.concurrent.{ExecutionContext, Future}
+
+case class CallToGoogleApiFailedException(googleProject: GoogleProject, clusterName:String, exceptionStatusCode: Int, errorMessage:String) extends LeoException(s"Call to Google API failed for $googleProject/$clusterName. Message: $errorMessage",exceptionStatusCode)
+
+class GoogleDataprocDAO(protected val dataprocConfig: DataprocConfig)(implicit val system: ActorSystem, val executionContext: ExecutionContext)
+  extends DataprocDAO with GoogleUtilities {
+
+  private val httpTransport = GoogleNetHttpTransport.newTrustedTransport
+  private val jsonFactory = JacksonFactory.getDefaultInstance
+  private val cloudPlatformScopes = List(PubsubScopes.CLOUD_PLATFORM)
+
+  private lazy val dataproc = {
+    new Dataproc.Builder(GoogleNetHttpTransport.newTrustedTransport,
+      JacksonFactory.getDefaultInstance, getDataProcServiceAccountCredential)
+      .setApplicationName("dataproc").build()
+  }
+  private def getDataProcServiceAccountCredential: Credential = {
+    new GoogleCredential.Builder()
+      .setTransport(httpTransport)
+      .setJsonFactory(jsonFactory)
+      .setServiceAccountId(dataprocConfig.serviceAccount)
+      .setServiceAccountScopes(cloudPlatformScopes.asJava)
+      .setServiceAccountPrivateKeyFromPemFile(new java.io.File(dataprocConfig.serviceAccountPemPath))
+      .build()
+  }
+
+
+  def createCluster(googleProject: String, clusterName: String, clusterRequest: ClusterRequest)(implicit executionContext: ExecutionContext): Future[ClusterResponse] = {
+    val op = build(googleProject, clusterName, clusterRequest)
+    op.map{op =>
+      val metadata = op.getMetadata
+      ClusterResponse(clusterName, googleProject, metadata.get("clusterUuid").toString, metadata.get("status").toString, metadata.get("description").toString, op.getName)}
+  }
+
+  private def build(googleProject: String, clusterName: String, clusterRequest: ClusterRequest)(implicit executionContext: ExecutionContext): Future[Operation] = {
+    Future {
+      //currently, the bucketPath of the clusterRequest are not used - it will be used later as a place to store notebooks and results
+      val metadata = clusterRequest.labels + ("docker-image" -> dataprocConfig.dataprocDockerImage)
+      val gce = new GceClusterConfig().setMetadata(metadata.asJava).setServiceAccount(clusterRequest.serviceAccount)
+      val initActions = Seq(new NodeInitializationAction().setExecutableFile(dataprocConfig.dataprocInitScriptURI))
+      val clusterConfig = new ClusterConfig().setGceClusterConfig(gce).setInitializationActions(initActions.asJava)
+      val cluster = new Cluster().setClusterName(clusterName).setConfig(clusterConfig)
+      val request = dataproc.projects().regions().clusters().create(googleProject, dataprocConfig.dataprocDefaultZone, cluster)
+      try {
+        executeGoogleRequest(request)
```

Comment:
switch our usages of `executeGoogleRequest` to something like `retryWhen500OrGoogleError`